test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -599,6 +600,22 @@ private static String groupName(ThreadGroup threadGroup) {
     }
 
     /**
+     * Builds a set of count unique items. Make sure that the items properly implement equals and hashcode.
+     */
+    public static <T> Set<T> randomUnique(Supplier<T> supplier, int count) {
+        Set<T> things = new HashSet<>();
+        int maxTries = count * 10;
+        for (int t = 0; t < maxTries; t++) {
+            if (things.size() == count) {
+                return things;
+            }
+            things.add(supplier.get());
+        }
+        // Oh well, we didn't get enough unique things. It'll be ok.
```

Comment:
Will do! Do you remember any other spots we do this and I can switch those spots to here?